### PR TITLE
WWT-4743 - Return proper error code on CONTRACT_REVERT

### PR DESCRIPTION
**Pull request by denshuliar**

### CODE REVIEW

File: src/services/ApiService.ts [hunk 1]
```diff
@@ -206,6 +206,7 @@ export const GET = async (network: Network, route: string) => {
                 .then((x) => x.json());
         } catch (e) {
             // console.log(`Mirror node service (${service.name}) failed to make request: ${service.url}${route}`);
+            throw new Error(`Mirror node service (${service.name}) failed to make request: ${service.url}${route}`);
```
Comment (Dubiy) [inline src/services/ApiService.ts]: please remove

Comment (Dubiy) [reply src/services/ApiService.ts]: otherwise it will fail on first record

File: src/services/ApiService.ts [hunk 2]
```diff
@@ -761,3 +762,48 @@ export const getNftMetadataFromIpfs = async (cid: string): Promise<NftMetadata>
 
     return response.json();
 };
+
+export const getContractErrorMessage = async (network: Network, param: string, contractId: string) => {
+    const MAX_ATTEMPTS = 5;
+    const INTERVAL_MS = 400;
```
Comment (Dubiy) [inline src/services/ApiService.ts]: why 400? 2000ms may not be enough

File: src/services/ApiService.ts [hunk 3]
```diff
@@ -761,3 +762,48 @@ export const getNftMetadataFromIpfs = async (cid: string): Promise<NftMetadata>
...
+                    } catch (error) {
+                        reject(error);
+                    }
+                }, INTERVAL_MS);
+            });
+
+            if (response && response.transactions && response.transactions.length > 0) {
+                const consensusTimestamp = response.transactions[0].consensus_timestamp;
+
+                let contractResponse: any = await new Promise((resolve, reject) => {
```
Comment (Dubiy) [inline src/services/ApiService.ts:785]: why timeout here also?

File: src/services/ApiService.ts [hunk 4]
```diff
@@ -761,3 +762,48 @@ export const getNftMetadataFromIpfs = async (cid: string): Promise<NftMetadata>
...
+                    }, INTERVAL_MS);
+                });
+
+                return contractResponse?.error_message;
+            } else {
+                throw new Error("No transactions found in the response");
+            }
+        } catch (error) {
+            if (attempt === MAX_ATTEMPTS) {
+                throw new Error(`Failed to fetch contract error message after ${MAX_ATTEMPTS} attempts`);
```
Comment (Dubiy) [inline src/services/ApiService.ts]: if we can't fetch error reason we need to skip this problem, and return to original flow  where CONTRACT_REVERT_EXECUTED without reason

File: test/src/index.test.ts [hunk 5]
```diff
@@ -330,6 +330,21 @@ test('bladeSdk.contractCallFunction', async () => {
         expect(result.error.reason.includes("CONTRACT_REVERT_EXECUTED")).toEqual(true);
     }
 
+    try {
+        // fail on wrong function params (CONTRACT_REVERT_EXECUTED) with error_message
+        params = new ParametersBuilder()
+        result = await bladeSdk.contractCallFunction(contractId, "revert_fnc", params, accountId, privateKey, 1000000, true, completionKey);
+        checkResult(result);
```
Comment (Dubiy) [inline test/src/index.test.ts]: we should check that contractCallFunction throws error. replace with

```
expect("Code should not reach here").toBeNull();
```

Comment (denshuliar) [reply test/src/index.test.ts]: Fixed ✅

File: src/services/ApiService.ts (hunk 2 above)
Comment (denshuliar) [reply src/services/ApiService.ts]: Fixed ✅

File: src/services/ApiService.ts (hunk 1 above)
Comment (denshuliar) [reply src/services/ApiService.ts]: Fixed ✅

File: src/services/ApiService.ts (hunk 3 above)
Comment (denshuliar) [reply src/services/ApiService.ts:785]: Fixed ✅

File: src/services/ApiService.ts (hunk 4 above)
Comment (denshuliar) [reply src/services/ApiService.ts]: Fixed ✅

**Review by Dubiy — APPROVED** [review]

**Pull request merged**